SourceCpp/React.cpp

```diff
@@ -146,30 +146,26 @@ PeleC::react_state(
           amrex::Real chemintg_cost;
           amrex::Real current_time = 0.0;
 
-          amrex::Real* rY_in;
-          amrex::Real* rY_src_in;
-          amrex::Real* re_in;
-          amrex::Real* re_src_in;
-
-#ifdef AMREX_USE_CUDA
-          int reactor_type = 1;
-          cudaError_t cuda_status = cudaSuccess;
-          cudaMallocManaged(
-            &rY_in, (NUM_SPECIES + 1) * ncells * sizeof(amrex::Real));
-          cudaMallocManaged(
-            &rY_src_in, NUM_SPECIES * ncells * sizeof(amrex::Real));
-          cudaMallocManaged(&re_in, ncells * sizeof(amrex::Real));
-          cudaMallocManaged(&re_src_in, ncells * sizeof(amrex::Real));
-
+          amrex::Vector<amrex::Real> h_rY_in(ncells * (NUM_SPECIES + 1));
+          amrex::Gpu::DeviceVector<amrex::Real> rY_in(
+            ncells * (NUM_SPECIES + 1));
+          amrex::Real* d_rY_in = rY_in.data();
+          amrex::Vector<amrex::Real> h_rY_src_in(ncells * NUM_SPECIES);
+          amrex::Gpu::DeviceVector<amrex::Real> rY_src_in(ncells * NUM_SPECIES);
+          amrex::Real* d_rY_src_in = rY_src_in.data();
+          amrex::Vector<amrex::Real> h_re_in(ncells);
+          amrex::Gpu::DeviceVector<amrex::Real> re_in(ncells);
+          amrex::Real* d_re_in = re_in.data();
+          amrex::Vector<amrex::Real> h_re_src_in(ncells);
+          amrex::Gpu::DeviceVector<amrex::Real> re_src_in(ncells);
+          amrex::Real* d_re_src_in = re_src_in.data();
+
```

Comment:
do we have to clear these at the end like h_rY_in.clear()?

Reply:
This is a good thought, but I think when they fall out of scope, their destructor does it. I will double check.

Reply:
I can't find any example of a `clear` happening in AMR-Wind and they don't mention it here https://amrex-codes.github.io/amrex/docs_html/GPU.html#gpu-vectors so I'm assuming all these vectors get cleared automatically even on device.

Reply:
I was hoping that will be the case.. you never know sometimes!

Reply:
Definitely. I will still need to do more testing of this by running longer simulations once it's merged.